Kernel/Devices/FloppyDiskDevice.cpp

```diff
@@ -124,7 +124,8 @@ bool FloppyDiskDevice::read_sectors_with_dma(u16 lba, u16 count, u8* outbuf)
 
     motor_enable(is_slave() ? 1 : 0); // Should I bother casting this?!
     write_ccr(0);
-    recalibrate(); // Recalibrate the drive
+    recalibrate();
+    ASSERT(seek(lba));
```

Comment:
Let's not do effectful things in `ASSERT` macros, since they will be compiled out in release builds.

Reply:
Detect it returned `false` and print an error?

Reply:
That'll work :)